src/pages/home/report/ReportActionItem.tsx

```diff
@@ -73,198 +76,219 @@ import ReportActionItemMessage from './ReportActionItemMessage';
+    action: OnyxTypes.ReportAction;
 
     /** Should the comment have the appearance of being grouped with the previous comment? */
-    displayAsGroup: PropTypes.bool.isRequired,
+    displayAsGroup: boolean;
 
     /** Is this the most recent IOU Action? */
-    isMostRecentIOUReportAction: PropTypes.bool.isRequired,
+    isMostRecentIOUReportAction: boolean;
 
     /** Should we display the new marker on top of the comment? */
-    shouldDisplayNewMarker: PropTypes.bool.isRequired,
+    shouldDisplayNewMarker: boolean;
 
     /** Determines if the avatar is displayed as a subscript (positioned lower than normal) */
-    shouldShowSubscriptAvatar: PropTypes.bool,
+    shouldShowSubscriptAvatar?: boolean;
 
     /** Position index of the report action in the overall report FlatList view */
-    index: PropTypes.number.isRequired,
-
-    /** Draft message - if this is set the comment is in 'edit' mode */
-    draftMessage: PropTypes.string,
-
-    /** Stores user's preferred skin tone */
-    preferredSkinTone: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-
-    ...windowDimensionsPropTypes,
-    emojiReactions: EmojiReactionsPropTypes,
-
-    /** IOU report for this action, if any */
-    iouReport: reportPropTypes,
+    index: number;
 
     /** Flag to show, hide the thread divider line */
-    shouldHideThreadDividerLine: PropTypes.bool,
-
-    /** The user's wallet account */
-    userWallet: userWalletPropTypes,
-
-    /** All the report actions belonging to the report's parent */
-    parentReportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
-};
-
-const defaultProps = {
-    draftMessage: undefined,
-    preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
-    emojiReactions: {},
-    shouldShowSubscriptAvatar: false,
-    iouReport: undefined,
-    shouldHideThreadDividerLine: false,
-    userWallet: {},
-    parentReportActions: {},
-};
-
-function ReportActionItem(props) {
+    shouldHideThreadDividerLine?: boolean;
+
+    linkedReportActionID?: string;
+} & ReportActionItemOnyxProps;
+
+const isIOUReport = (actionObj: OnyxEntry<OnyxTypes.ReportAction>): actionObj is ReportActionBase & OriginalMessageIOU => actionObj?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU;
+
+function ReportActionItem({
+    action,
+    report,
+    linkedReportActionID,
+    displayAsGroup,
+    emojiReactions,
+    index,
+    iouReport,
+    isMostRecentIOUReportAction,
+    parentReportActions,
+    preferredSkinTone = CONST.EMOJI_DEFAULT_SKIN_TONE,
+    shouldDisplayNewMarker,
+    userWallet,
+    shouldHideThreadDividerLine = false,
+    shouldShowSubscriptAvatar = false,
+    policyReportFields,
+}: ReportActionItemProps) {
+    const {translate} = useLocalize();
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const blockedFromConcierge = useBlockedFromConcierge();
+    // TODO need to fix createOnyxContext to report types as OnyxCollection if provided key is collection
```

Comment:
`useReportActionsDrafts` is returning `OnyxTypes.ReportActionsDrafts` but in fact its returning `OnyxCollection<OnyxTypes,ReportActionsDrafts>`

Reply:
@blazejkustra suggested to move it to `withOnyx` but its not working, it breaking funcionality of Editing messages

Reply:
`useReportActionsDrafts` is created in `src/components/createOnyxContext.tsx`, a generic function to create HOCs, providers and hooks based on a Onyx key, so there is the place where the typings would have to be fixed.

Reply:
Its fixed now, I adjusted type in ONYXKEYS and in migrations files and now its good